luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/DefaultLogicalTableGroupFactory.java

```diff
@@ -0,0 +1,146 @@
+import com.yahoo.bard.webservice.data.metric.MetricDictionary;
+import com.yahoo.bard.webservice.data.time.Granularity;
+import com.yahoo.bard.webservice.data.time.GranularityDictionary;
+import com.yahoo.bard.webservice.data.time.GranularityParser;
+import com.yahoo.bard.webservice.data.time.StandardGranularityParser;
+import com.yahoo.bard.webservice.table.LogicalTable;
+import com.yahoo.bard.webservice.table.PhysicalTable;
+import com.yahoo.bard.webservice.table.TableGroup;
+import com.yahoo.bard.webservice.table.TableIdentifier;
+import com.yahoo.bard.webservice.util.GranularityParseException;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Period;
+import org.joda.time.ReadablePeriod;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedHashSet;
+
+/**
+ * A factory that is used by default to support LogicalTableGroup.
+ *
+ * A LogicalTableGroup is essentially a map from TableIdentifier to LogicalTable.
+ * This concept exists because one configuration string (name) defines multiple LogicalTableIdentifiers
+ * (name, granularity pair). LogicalTableGroup allows us to collect all the distinct LogicalTables for a single name,
+ * mapping LogicalTableIdentifier to LogicalTable.
+ *
+ * At the external configuration layer, configuration users will define a logical table group (keyed by name)
+ * with a list of granularities. The factory will produce the collection of LogicalTables keyed by identifier
+ * (name, grain pairs), to be merged into the table dictionary.
+ */
+public class DefaultLogicalTableGroupFactory implements Factory<LogicalTableGroup> {
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultLogicalTableGroupFactory.class);
+    private static final String ENTITY_TYPE = "default logical table group";
+    private static final String GRANULARITY_PARSING_ERROR =
+            "granularity: %s does not exist in the granularityDictionary";
+    private static final String GRANULARITY_DICTIONARY_MISSING = "granularityDictionary missing from the " +
+            "LuthierIndustrialPark. Will not build " + ENTITY_TYPE + ": %s.";
+
+    /**
+     * Build a group of LogicalTable, it has one LogicalTable per granularity.
+     *
+     * @param name  apiName of the LogicalTable
+     * @param configTable  the LogicalTableGroup's configuration
+     * @param resourceFactories  the source for locating dependent objects
+     *
+     * @return  A newly constructed config instance for the name and config provided
+     */
+    @Override
+    public LogicalTableGroup build(String name, ObjectNode configTable, LuthierIndustrialPark resourceFactories) {
+        LogicalTableGroup logicalTableGroup = new LogicalTableGroup();
+        GranularityDictionary granularityDictionary = resourceFactories.getGranularityDictionary();
+        if (granularityDictionary == null) {
+            LOG.warn(String.format(GRANULARITY_DICTIONARY_MISSING, name));
+            return logicalTableGroup;
```

Comment:
I would just explode here, with the error message what you're
currently logging as a warning. An empty table group is pretty useless, and
will lead to weird errors people won't notice until runtime, and that they'll
have to triage.

Reply:
This is not exploding since it makes previously passing unit tests unusable. (As they are initialized without a granularity dictionary, nothing will ever pass)

You concern is indeed interesting, though. I can go about and refactor all previous and future test to only use a LuthierBinderFactory to initialize the LIP, if you insist on stalling the entire process when seeing a null value.

Reply:
Creating a LuthierBinderFactory for all the tests sounds like massive overkill.

Why not just initialize their LuthierIndustrialParks with a GranularityDictionary? 

If you haven't already, you should create a utility for your tests `LuthierIndustrialPark initializeLuthierIndustrialParkBuilder` which builds a `LuthierIndustrialParkBuilder` with whatever configuration you're always going to need. 

Then, anytime we need to add additional state to the park, you can update that method rather than having to modify all your tests.

Furthermore, by returning the builder, it's easy enough for tests to attach their own custom configuration as well.